internal/api/dto/plan.go

```diff
@@ -43,6 +48,131 @@ func (r *CreatePlanRequest) Validate() error {
 		}
 	}
 
+	for _, cg := range r.CreditGrants {
+		if err := r.validateCreditGrantForPlan(cg); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// validateCreditGrantForPlan validates a credit grant for plan creation
+// This is similar to CreditGrant.Validate() but skips plan_id validation since
+// the plan ID will be set after the plan is created
+func (r *CreatePlanRequest) validateCreditGrantForPlan(cg CreateCreditGrantRequest) error {
```

Comment:
TODO: later - reuse the validation part as most of it is common from subscription creation